src/main/java/com/customerApi/api/services/CustomerService.java

```diff
@@ -0,0 +1,150 @@
+package com.customerApi.api.services;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.hibernate.exception.JDBCConnectionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.customerApi.api.exception.CustomerNotFoundException;
+import com.customerApi.api.modals.Customer;
+import com.customerApi.api.repositories.CustomerRepository;
+
+@Service
+public class CustomerService {
+
+	private static final Logger logger = LoggerFactory.getLogger(CustomerService.class);
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	public Optional<Integer> addCustomerData(Customer customer) {
+
+		Optional<Integer> customerId = null;
+
+		try {
+
+			Customer cust = customerRepo.save(customer);
+
+			customerId = Optional.of(cust.getId());
+
+		} catch (JDBCConnectionException e) {
+			logger.error("Database connection error" + e);
+
+		} catch (Exception e) {
+			logger.error("Customer {} record has not saved Successfully " + e);
+		}
+
+		if (!customerId.isPresent()) {
+
+			logger.error("Customer ID {}" + customerId);
+			throw new CustomerNotFoundException(" Customer id-" + customerId);
+		}
+		logger.trace("Customer ID {}" + customerId);
+		return customerId;
+
+	}
+
+	public List<Customer> getCustomersData() {
+		List<Customer> allCustomer = null;
+
+		try {
+			allCustomer = customerRepo.findAll();
+			Optional.of(allCustomer);
+
+		} catch (JDBCConnectionException e) {
+			logger.error("Database connection error" + e);
+
+		} catch (Exception e) {
+			logger.error("Database connection error " + e);
```

Comment:
In case of this error why you are just printing the error and let the flow move forward, you need to throw exception here right away and pass a meaning message to the front end. You should throw proper exception here otherwise frontend will receive empty data and it will think there is no customers at backend. But real reason is database is down.